### ISSUE TITLE
'load data' problem from file whose lines separated by unicode string

### ISSUE THREAD

**Issue by hustwk** [pingcap/tidb#7188]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

The file 000000_2 contains fields terminated by unicode '\u0017', then

load data local infile '/root/test/test_table/000000_2' into table test.test_table fields TERMINATED by X'\11';
or
load data local infile '/root/test/test_table/000000_2' into table test.test_table fields TERMINATED by '\021';
or
load data local infile '/root/test/test_table/000000_2' into table test.test_table fields TERMINATED by '\u0017';

2. What did you expect to see?
like mysql , load data successfully.

3. What did you see instead?
ERROR 1105 (HY000): line 1 column 125 near "" (total length 125)
or
ERROR 2013 (HY000): Lost connection to MySQL server during query

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
tidb:v2.0.5

**Comment by zimulala** [comment]
@hustwk 
Thanks for your report!
```
tidb> load data local infile '/tmp/x.log' into table test.test_table fields TERMINATED by '\11';
Query OK, 0 rows affected (0.00 sec)
tidb>  load data local infile '/tmp/x.log' into table test.test_table fields TERMINATED by '\021';
Query OK, 0 rows affected (0.01 sec)
tidb> load data local infile '/tmp/x.log' into table test.test_table fields TERMINATED by '\u0017';
Query OK, 0 rows affected (0.00 sec)
```
Could you show me the content of `000000_2` and the result of `show create table test.test_table`?

**Comment by hustwk** [comment]
The simplest steps to reproducing the error.

step 1: create test_table
`create database test;
use test;
CREATE TABLE `test_table` (
  `key` varchar(12) NOT NULL,
  `value` varchar(8) NOT NULL
);`

step2: generate test.data, run bash command
`printf key'\x11'value > test.data`

check the column separator is 0x11 in hex or 021 in octal format:
`# cat test.data |hexdump -c          
0000000   k   e   y 021   v   a   l   u   e   n    `

step3: load test.data
`load data local infile '/root/test.data' into table test.test_table fields TERMINATED by X'11';
ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
Connection id:    14
Current database: test

ERROR 1105 (HY000): line 1 column 100 near "" (total length 100)

mysql> load data local infile '/root/test.data' into table test.test_table fields TERMINATED by '\11';
ERROR 2013 (HY000): Lost connection to MySQL server during query

mysql> load data local infile '/root/test.data' into table test.test_table fields TERMINATED by '\021';
ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
Connection id:    16
Current database: test

ERROR 2013 (HY000): Lost connection to MySQL server during query

mysql> load data local infile '/root/test.data' into table test.test_table fields TERMINATED by '\u0017';
ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
Connection id:    17
Current database: test

ERROR 2013 (HY000): Lost connection to MySQL server during query

mysql> select * from test_table;
ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
Connection id:    18
Current database: test

Empty set (0.02 sec)

mysql> select * from test_table;
Empty set (0.00 sec)
`

Step 4: While the same command executed in Mysql:
`mysql> load data local infile '/root/test.data' into table test.test_table fields TERMINATED by X'11';
Query OK, 1 row affected (0.00 sec)
Records: 1  Deleted: 0  Skipped: 0  Warnings: 0

mysql> select * from test_table;
+-----------+-------+
| key       | value |
+-----------+-------+
| key       | value |
+-----------+-------+`

So tidb can not load data terminated by unicode '\021' or '11' or '\u0017' while Mysql can do it.

**Comment by hustwk** [comment]
Sine the default delimiter for Hive table files is **^A(ctrl-a)**, which is the **unicode '\u0001'**. So if we want to load existed hive tables files directly to TiDB, we should specify the fields **unicode** separator.
This may be common usage scenarios, and may have some best practices. 
Do you have any suggestion?

**Comment by hustwk** [comment]
Three working days have passed, it seems that no one cares about this problem while I gave detailed description to reproduce the problem. 
The community is not so active, but I would be very grateful if you can care about user’s questions.

**Comment by shenli** [comment]
@hustwk We are working on a release in recent days. We will investigate your issue soon.

**Comment by shenli** [comment]
@hustwk I tried your case. But got an error:

> mysql> load data local infile '/tmp/test.data' into table test.test_table fields TERMINATED by X'11';
ERROR 1105 (HY000): line 1 column 93 near "" (total length 93)
mysql>

**Comment by jackysp** [comment]
Sorry for the late reply @hustwk . We have a plan to improve the starting/terminated/separated/enclosed in load data in the future. This the unicode will be supported then.

**Comment by hustwk** [comment]
@shenli Thanks for your reply. 
That is the problem. The same SQL can be executed correctly in MySQL while can **NOT** executed in TiDB. So we can not load files with unicode separator such as existed hive table files.

**Comment by shenli** [comment]
@hustwk Thanks! I have reproduced the issue. We will fix it.

**Comment by hustwk** [comment]
@jackysp Thanks for your reply. Glad to hear that and thanks for all your contribute!

**Comment by kolbe** [comment]
It looks like this is resolved now in the master branch (c0d6185144e25c66c3587b5300b1756f965b553b).

The '\11', '\011', and '\u0011' escapes don't do anything in MySQL, so they don't work there either. But parsing `TERMINATED BY x'11'` and the importation of the file both work correctly in master.

**Comment by nullnotnil** [comment]
Confirming that this has been fixed:
```sql
mysql> load data local infile '/tmp/test.data' into table test_table fields TERMINATED by X'11';
Query OK, 1 row affected (0.03 sec)
Records: 1  Deleted: 0  Skipped: 0  Warnings: 0

mysql> select * from test_table;
+-----+-------+
| key | value |
+-----+-------+
| key | value |
+-----+-------+
1 row in set (0.01 sec)
```

I am going to close this issue now. Thanks!
